osu.Game.Rulesets.Osu/Mods/OsuModSnapColour.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable disable
+
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.ObjectExtensions;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Screens.Edit;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    /// <summary>
+    /// Mod that colours <see cref="HitObject"/>s based on the musical division they are on
+    /// </summary>
+    public class OsuModSnapColour : ModSnapColour, IApplicableToBeatmap, IApplicableToDrawableHitObject
+    {
+        [Resolved]
+        private OsuColour colours { get; set; } = new OsuColour();
```

Comment:
This is not how Dependency Injection works. You don't need to initialise the auto-property, with NRT enabled use the null-forgiving operator (see https://github.com/ppy/osu/wiki/Nullable-Reference-Types-(NRT)#dependency-injection-considerations)

Reply:
Thanks, I removed the dependency injection and initialized OsuColour as readonly. 

I could be wrong but it seems like the OsuColour dependency injection was implemented for Drawable classes so it needed to be initialized in this class

Reply:
Yes, but you don't need to initialise the auto-property in such cases. You only need to initialise it to `null! because of NRT`. Dependencies will be provided automatically when the drawable will be loaded.